pkg/kubelet/server/stats/summary.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package stats
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	cadvisorapiv2 "github.com/google/cadvisor/info/v2"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/kubelet/cm"
+	"k8s.io/kubernetes/pkg/kubelet/dockertools"
+	"k8s.io/kubernetes/pkg/kubelet/leaky"
+)
+
+// buildSummary aggregates the data passed as arguments and parses it into a *Summary structure
+func buildSummary(
+	node *api.Node,
+	nodeConfig cm.NodeConfig,
+	infos map[string]cadvisorapiv2.ContainerInfo) (*Summary, error) {
+
+	rootInfo, found := infos["/"]
+	if !found || len(rootInfo.Stats) < 1 || latestContainerStats(&rootInfo) == nil {
+		return nil, fmt.Errorf("Missing stats for root container")
+	}
+
+	rootStats := containerInfoV2ToStats("", &rootInfo)
+	nodeStats := NodeStats{
+		NodeName: node.Name,
+		CPU:      rootStats.CPU,
+		Memory:   rootStats.Memory,
+		Network:  containerInfoV2ToNetworkStats(&rootInfo),
+	}
+
+	systemContainers := map[string]string{
+		SystemContainerKubelet: nodeConfig.KubeletContainerName,
```

Comment:
I just saw this, not sure why it fell through my e-mail cracks, but on systemd, the kubelet container name is meaningless as provided in `NodeConfig` since when launched on systemd, it inherently is already in a container.  For that use case, we need to have a bootstrapping module report back its actual container name.

Linking this PR so I don't miss this use case.
https://github.com/kubernetes/kubernetes/pull/20748